src/server/json_family.cc

```diff
@@ -778,6 +778,39 @@ OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, s
   return vec;
 }
 
+// Returns numeric vector that represents the new length of the array at each path, or Null reply
+// if the matching JSON value is not an array.
+OpResult<vector<OptSizeT>> OpArrAppend(const OpArgs& op_args, string_view key, string_view path,
+                                       const vector<json>& append_values) {
+  vector<OptSizeT> vec;
+
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  auto cb = [&](const string& path, json& val) {
+    if (!val.is_array()) {
+      vec.emplace_back(nullopt);
+      return;
+    }
+    for (auto& new_val : append_values) {
+      val.emplace_back(new_val);
+    }
+    vec.emplace_back(val.size());
+  };
+
+  json j = result.value();
```

Comment:
Just a side note: Doesn't this use the copy constructor and not the move one? (in all occurrences)

Reply:
Yes, seems you found a performance bug.

Reply:
Yes, it should be fixed.